apps/src/blockly/addons/cdoXml.js

```diff
@@ -74,6 +75,29 @@ export default function initializeBlocklyXml(blocklyWrapper) {
   blocklyWrapper.Xml.createBlockOrderMap = createBlockOrderMap;
 }
 
+/**
+ * Adds a mutation element to a block if it should have an open miniflyout.
+ * CDO Blockly uses an unsupported method for serializing miniflyout state
+ * where arbitrary block attribute could be used to manage extra state.
+ * Mainline Blockly expects a mutator. The precence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ * @returns {Map} A map with partitioned block index as key and original index in the XML as value.
+ */
+export function addMutationToMinitoolboxBlocks(blockElement) {
+  const miniflyoutAttribute = blockElement.getAttribute('miniflyout');
+  if (!miniflyoutAttribute) {
+    return; // Not a miniflyout block
+  }
+  const mutationElement = blockElement.ownerDocument.createElement('mutation');
+  // The default icon is a '+' symbol which represents a currently-closed flyout.
+  const useDefaultIcon = miniflyoutAttribute === 'open' ? 'false' : 'true';
```

Comment:
Does this need to be a string and not a boolean?

Reply:
I don't think it matters technically. With `setAttribute`, any non-string value is converted automatically into a string. I went with strings here because when we get these values later, we expect them to be strings.

https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute